discovery-provider/src/queries/get_tracks.py

```diff
@@ -131,6 +132,13 @@ def _get_tracks(session, args):
             )
         )
 
+    if "filter_tracks" in args and args.get("filter_tracks") != "all":
+        filter_tracks = args.get("filter_tracks")
+        if filter_tracks == "unlisted":
+            base_query = base_query.filter(TrackWithAggregates.is_unlisted == True)
+        else:
+            base_query = base_query.filter(TrackWithAggregates.is_unlisted == False)
+
```

Comment:
This logic probably conflicts with the `is_unlisted` filter appended up on line 90.

I think I'd combine it with the line 90 logic, something like:

```
filter_tracks = args.get("filter_tracks")
if not is_authed_user:
    base_query = base_query.filter(TrackWithAggregates.is_unlisted == False)
elif is_authed_user and filter_tracks == "public":
    base_query = base_query.filter(TrackWithAggregates.is_unlisted == False)
elif is_authed_user and filter_tracks == "unlisted":
    base_query = base_query.filter(TrackWithAggregates.is_unlisted == True)
elif is_authed_user and filter_tracks == "all":
    pass
else:
    raise Exception('unreachable')
```

Reply:
I separated the two bc they're conceptually different; line 90 logic returns tracks the user is authorized to view, then separately the tracks are filtered according to the `filter_tracks` query params. If the query is not from an authed user and they request `filter_tracks=unlisted`, they just get 0 results.

Combining the logic would mean a query from an unauthed user with `filter_tracks=unlisted` would result in a list of public tracks, which would seem like the query param was just ignored if you didn't know about the authentication logic that we consider behind the scenes.

I suppose its more of a product question. id prefer keeping the filter track logic separate but let me know what you think

Reply:
I think @michellebrier is right!

Reply:
Okay yeah that makes sense to me now!